### DISCUSSION TITLE
How can I create a multiple images selection with Expo?

### DISCUSSION THREAD

**Discussion by doougui** [expo/expo#15210] [Archived: Questions]
So, I am new to RN/Expo and I am building an application with Expo and need to access the user camera roll (or assets) to select multiple images (not just one image as expo's ImagePicker does). 

So, since [CameraRoll from React Native](https://reactnative.dev/docs/0.60/cameraroll) is deprecated, what I was trying to do is to use @react-native-community/cameraroll but with no success yet, even tough my project is using the bare workflow.

Most of the posts I see give instructions to use expo's `MediaLibrary` with `getAssetsAsync` instead of `CameraRoll`, but the problem with this is that the image listing doesn't really happen as expected. I expect to be able to scroll all the way to the bottom of the "images selection box" without having to stop to "fetch the next page". Something like WhatsApp, Telegram, and popular apps like these does.

Something like this:

[![IMAGE ALT TEXT HERE](https://img.youtube.com/vi/w8FudJCQH3s/0.jpg)](https://www.youtube.com/watch?v=w8FudJCQH3s)

**Comment by keith-kurak** [comment]
You may find a native module that supports this; however, it does appear that needing to fetch the images in batches is a limitation of the native API's available. For instance, [react-native-multiple-image-picker](https://github.com/baronha/react-native-multiple-image-picker) (I've only looked this up, haven't tried it myself) appears to work by wrapping a cocoapod that includes a custom native view controller, which in turn fetches images in batches (but also tries to prefetch them before a certain scroll threshold is reached). With just `MediaLibrary` and RN list components, you may be able to achieve similar results with a good prefetch algorithm.
